src/energy_grid.F90

```diff
@@ -127,16 +125,6 @@ subroutine add_grid_points(list, energy)
     i = 1
     n = size(energy)
 
-    ! If the original list is empty, we need to allocate the first element and
-    ! store first energy point
-    if (.not. associated(list)) then
```

Comment:
It doesn't need to be allocated because it's no longer a pointer. I removed the do loop for a different reason: it's a bit redundant since the logic in the `if (current > list % size()) then` block below achieves the same effect.